### PR TITLE
[MM-51445] Replace compass-components icon component with compass icons

### PR DESCRIPTION
**Pull request by AshishDhama**
#### Summary
Replace compass-components icon component with compass icons

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-51445

#### Screenshots
Insights Icon
Screenshot 2023-03-30 at 3 31 10 PM

All the icons in the menu and Selected product icon in the nav bar
Screenshot 2023-03-30 at 3 31 02 PM

Edit Icon after editing the post
Screenshot 2023-03-30 at 3 30 40 PM

Market place icon in app bar
Screenshot 2023-03-30 at 4 41 11 PM

To enable app bar
Screenshot 2023-03-30 at 4 33 52 PM

#### Release Note
```release-note

- Replace compass-components icon component with compass icons

```

### CODE REVIEW

**Review by vish9812 — APPROVED** [review]

**Review by kostaspt — APPROVED** [review]

**Review by furqanmlk — COMMENTED** [review]
@AshishDhama, Can you please provide figma or QA steps?

File: webapp/channels/src/components/global_header/left_controls/product_menu/product_menu_list/product_menu_list.tsx [hunk 1]
```diff
@@ -128,25 +134,15 @@ const ProductMenuList = (props: Props): JSX.Element | null => {
                                 )}
                             </>
                         )}
-                        icon={
-                            <Icon
-                                size={16}
-                                glyph={'application-cog'}
-                            />
-                        }
+                        icon={<ApplicationCogIcon size={18}/>}
```
Comment (vish9812) [inline webapp/channels/src/components/global_header/left_controls/product_menu/product_menu_list/product_menu_list.tsx:137]: Is it okay to change the size from `16` to `18`?

Comment (AshishDhama) [reply webapp/channels/src/components/global_header/left_controls/product_menu/product_menu_list/product_menu_list.tsx:137]: Yeah, I checked the sizes were not same in HTML

Comment (AshishDhama) [reply webapp/channels/src/components/global_header/left_controls/product_menu/product_menu_list/product_menu_list.tsx:137]: I had to bump size in for all Icons

**Review by kostaspt — DISMISSED** [review]

**Review by vish9812 — APPROVED** [review]

**Review by furqanmlk — APPROVED** [review]

**Pull request merged**